helios-services/src/main/java/com/spotify/helios/master/ZooKeeperMasterModel.java

```diff
@@ -1056,6 +1056,15 @@ public Job removeJob(final JobId id, final String token)
       throw new HeliosRuntimeException("removing job " + id + " failed", e);
     }
 
+    // Delete job history on a best effort basis
+    try {
+      client.deleteRecursive(Paths.historyJob(id));
```

Comment:
It's possible, but there's no recursive delete in ZK so first have to do a recursive list outside the transaction and then add a delete for each node to the transaction. This seemed like more work - and the recursive delete after the fact seemed good enough so I opted for being lazy.